src/components/SkiaBadge.tsx

```diff
@@ -0,0 +1,167 @@
+import { StyleSheet } from 'react-native';
+import { Canvas, Group, Rect, RoundedRect, Shadow, SkParagraph, rect, rrect } from '@shopify/react-native-skia';
+import { memo, useCallback } from 'react';
+import Animated, { useAnimatedStyle, useDerivedValue, useSharedValue } from 'react-native-reanimated';
+import { SkiaText, useColorMode } from '@/design-system';
+import { TextSize } from '@/design-system/typography/typeHierarchy';
+import { TextWeight } from '@/design-system/components/Text/Text';
+import { TextColor } from '@/design-system/color/palettes';
+import { CustomColor } from '@/design-system/color/useForegroundColor';
+
+type Shadow = {
+  dx: number;
+  dy: number;
+  blur: number;
+  color: string;
+};
+
+type SkiaBadgeProps = {
+  text: string;
+  height?: number;
+  horizontalPadding?: number;
+  fillColor: string | string[];
+  textColor: TextColor | CustomColor;
+  strokeColor?: string | string[];
+  strokeWidth?: number;
+  dropShadows?: Shadow[];
+  innerShadows?: Shadow[];
+  fontSize?: TextSize;
+  fontWeight?: TextWeight;
+  lineHeight?: number;
+};
+
+export const SkiaBadge = memo(function SkiaBadge({
+  text,
+  height = 27,
+  horizontalPadding = 8,
+  fillColor,
+  textColor,
+  strokeColor,
+  strokeWidth = 2,
+  dropShadows = [],
+  innerShadows = [],
```

Comment:
I'd use undefined here and optional chain the access as these are unnecessarily created every time this function runs